### PR TITLE
Only use import.meta when strictly necessary

### LINKED ISSUE #8729: EXPORT_ES6 mode import.meta.url is not supported
Issue:
[A recent PR](https://github.com/emscripten-core/emscripten/pull/8306/files#diff-ae40e645f84e893ce5d037b5efc53be6R2850) resulted in the use of import.meta.url for EXPORT_ES6. However this functionality is currently only at stage 3 (proposal) which means support for this functionality is fairly low which includes eslint, babel, Firefox and Edge all not supporting it currently (all implement at stage 4 or later)

Context:
Looks like there was already a discussion about this and how support isn't quite there yet.
https://github.com/emscripten-core/emscripten/issues/7137

### PR DESCRIPTION
**Pull request by VirtualTim**
A webworker ES6 module doesn't have access to 'document.currentScript'. 'import.meta' was the only way I could see to achieve that functionality.

Fixes #8729.

Since this appears to fix the issue for everyone who commented on #8729, and no one proposed any better ideas, this is my suggestion.

It would be nice to use import.meta everywhere, and browsers seem to have good support, however the tooling appears to be lagging a bit behind.

### CODE REVIEW

**Comment by VirtualTim** [comment]
The one test failure is `test_glgears`. Looks like a flaky test.

**Comment by curiousdannii** [comment]
Have you tested with WASM=0? Won't it need it for the memory file?

You should also be able to test for SINGLE_FILE=0.

File: emcc.py [hunk 1]
```diff
@@ -2945,7 +2945,7 @@ def modularize():
       # after document.currentScript is gone, so we save it.
       # (when MODULARIZE_INSTANCE, an instance is created
       # immediately anyhow, like in non-modularize mode)
-      if shared.Settings.EXPORT_ES6:
+      if shared.Settings.EXPORT_ES6 and shared.Settings.USE_PTHREADS:
```
Comment (kripken) [inline emcc.py]: please add a comment explaining why pthreads are special

**Comment by kripken** [comment]
I don't quite understand this (sorry, I couldn't follow the discussion in the other issue either). Can you please summarize, assuming a reader that isn't familiar with ES6? :)

> It would be nice to use import.meta everywhere, and browsers seem to have good support, however the tooling appears to be lagging a bit behind.

What do you mean by "tooling" here? Which tools?

**Comment by k2snowman69** [comment]
> 
> 
> I don't quite understand this (sorry, I couldn't follow the discussion in the other issue either). Can you please summarize, assuming a reader that isn't familiar with ES6? :)
> 
> > It would be nice to use import.meta everywhere, and browsers seem to have good support, however the tooling appears to be lagging a bit behind.
> 
> What do you mean by "tooling" here? Which tools?

I'm the one who brought up tooling so I can chime in here... Tooling is mostly build/lint tools for the javascript ecosystem. For example:
 - Babel core does not support this functionality yet so you explicitly have to add the @babel/plugin-syntax-import-meta
 - Webpack is the same and requires @open-wc/webpack-import-meta-loader

And this is mainly because much of the javascript tooling that exists state that they only support TC39 stage 4 proposals and higher where import.meta.url is [still at stage 3](https://github.com/tc39/proposal-import-meta/#importmeta). This means that technically import.meta.url isn't even part of the ECMAScript spec yet, it's in the finalization process and once it gets to Stage 4 it will become part of the spec the following year. Browsers just like the feature enough that they have implemented it early but not everyone has.

Yes there are people who provide plugins, polyfills and extensions to support many pieces of this new functionality but without these additions (and emscripten documenting that these additions are now required), the core tooling will not compile them.

File: emcc.py (hunk 1 above)
Comment (VirtualTim) [reply emcc.py]: Done.

**Comment by kripken** [comment]
@k2snowman69 interesting, thanks!

This looks ok to me, aside from the test failures.

**Comment by kripken** [comment]
Great, thanks @VirtualTim!

**Pull request merged**